[IV-22-23] Objetivo 3 de @pedromarting3

pedrooot opened the pull request:
* [x]  ¿Incluye el fichero `iv.yaml` las claves de todos los objetivos
anteriores?
* [x]  ¿Se ha desarrollado el fichero para las tareas a partir de un
   issue?
* [x]  ¿Ese issue está asignado a un milestone?
* [x]  ¿Hemos justificado correctamente en caso de elegir un gestor de
   dependencias el que se use como gestor de tareas?
* [x]  ¿El PR, que también es un issue, está asignado al mismo milestone?

pedrooot committed:
closes #15 creando iv.yaml

pedrooot committed:
Avanza #13 creando package.json

pedrooot committed:
Avanza #14 creando Gruntfile.js

pedrooot committed:
Avanza #13 añadiendo version de node a package.json

pedrooot committed:
avanza #14 modificando Gruntfile.js

pedrooot committed:
closes #13 #14 #15 updating Readme.md

pedrooot committed:
closes #15 updating iv.yaml

pedrooot committed:
closes #15 updating iv.yaml

pedrooot committed:
closes #14 creanting gestor_tareas.md and gestor_dependencias.md

pedromarting3 commented:
@JJ listo para revisión

arsa-dev commented on Gruntfile.js:
```diff
@@ -0,0 +1,27 @@
+
+const { default: nodeTest } = require("node:test");
+
+//WIP
+module.exports = function (grunt) {
+  grunt.initConfig({
+    pkg: grunt.file.readJSON("package.json"),
+    check: {
+      build: {
+        src: ["src/**/*.js"],
+        dest: "dist/<%= pkg.name %>.js",
+      },
+    },
+    log: {
+      foo: [1, 2, 3],
+      bar: "Welcome to tutorialspoint",
+      sap: true,
+    },
+  });
+  grunt.registerTask("check", "check code", function () {
+    "node --check src/*.js";
+  });
```
Revisa el comentario que ya no es WIP, además creo que hay tareas ejemplo que realmente no se usan y no están aplicando a nada del PMV por tanto no deberían de incluirse.

arsa-dev commented on README.md:
````diff
@@ -20,4 +20,10 @@ Los principales clientes del servicio serán:
\ No newline at end of file
+* [Historias de usuario](/docs/Historias.md)
+
+## Check format
+Para comprobar el formato de los directorios ejecuta el siguiente comando:
+```shell
+grunt check
+```
````
Mas que formato sería sintaxis, para formato sería un linter

arsa-dev reviewed (changes requested):
Las justificaciónes para la toma de decisiones las veo correctas, así como la implementación de la tarea check, lo único esos detalles que te comento.

pedrooot committed:
closes #14 updating Gruntfile.js

pedrooot committed:
closes #13 updating package.json

pedrooot committed:
update Readme.md

pedromarting3 commented:
@arsa-dev todo hecho, gracias por la revisión.🚀🏌🏽‍♂️

JJ commented on Gruntfile.js:
```diff
@@ -0,0 +1,7 @@
+const { default: nodeTest } = require("node:test");
```
Esto no lo estás usando. Además, requiere una versión específica de node que no estás especificando en ningún lado.

JJ commented on docs/gestor_dependencias.md:
```diff
@@ -0,0 +1,6 @@
+A la hora de escoger el gestor de dependencias, he barajado diferentes posibilidades:  
+* npm-> (Node Package Management) es un gestor de dependencias  desarrollado sobre Node.js en el que cada proyecto puede utilizar el archivo de instalación package.json a través de NMP y gestionarlo con GULP. Las dependencias se pueden optimizar y actualizar desde la terminal. Se pueden desarrollar nuevos proyectos desde archivos de dependencias y números de versión insertados automáticamente des ...
+* bun -> se basa en la velocidad, rendimiento y polifacético (empaquetador, administrador de paquetes...). El objetivo de BUN es ejecutar la mayor parte de JavaScript fuera de los navegadores para mejorar el rendimiento y complejidad a su infraestructura futura.
+
```
Hay más posibilidades, que además se usan de forma habitual (y hemos comentado en clase). Comenta por qué las has descartado.

JJ commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,6 @@
+# Elección del gestor de tareas que se usará
+A la hora de escoger el gestor de tareas, he barajado diferentes posibilidades:  
```
Lo que quiero que me contéis son las posibilidades que *hay*, no las que habéis barajado. He comentado uno más en clase, que no estás incluyendo aquí.

Sería conveniente también que añadieras algún criterio por el que has usado uno u otro; por ejemplo, siendo bun útil para las dos cosas, cabría pensar que es una buena elección.

JJ reviewed (changes requested):
Como os he comentado en clase, lo esencial es que busquéis cuales son las buenas prácticas habituales en el lenguaje de programación con el que estéis trabajando e incluyáis todas las posibilidades.

JJ committed:
Aviso para navegantes

Es el primero que se ha entregado, así que siempre se corre este peligro. El que avisa no es traidor, es avisador.

pedrooot committed:
closes #14 updating Gruntfile.js

pedrooot committed:
Merge pull request #18 from JJ/patch-1

Aviso para navegantes

pedrooot committed:
closes #13 updating gestor_dependencias.md

pedrooot committed:
closes #14 updating gestor_tareas.md

pedromarting3 commented:
@JJ listo para revisión.

JJ commented on package.json:
```diff
@@ -0,0 +1,16 @@
+    "grunt-contrib-nodeunit": "~0.4.1",
+    "grunt-contrib-uglify": "^0.5.1",
+    "grunt-contrib-watch": "^1.1.0",
+    "node": "~18.11.0"
```
¿Seguro que te hace falta esto? ¿Las últimas versiones de npm te instalan también node? En todo caso, ¿no sería una dependencia y no una dependencia de desarrollo?

JJ reviewed (approved):
Meter node como dependencia no lo he visto nunca; typescript sí, pero no node. Salvo eso, :+1:

pedrooot committed:
closes #13 eliminando node en package.json
